### PR TITLE
FED Errors vs FED ID 2D plot

### PR DESCRIPTION
**Pull request by threus**
Added 2D ME counting Strip FED errors for each FED ID.
Saved into SiStrip/ReadoutView.

### CODE REVIEW

**Comment by dmitrijus** [comment]
please test

Code looks fine. I will +1 after the tests.

**Comment by threus** [comment]
Is it a legitimate error? I had no crashes running whiteRabbit on Run1 data. Could you help me to understand what's wrong with setBinLabel, when it works for other workflows? Thanks.

**Comment by davidlange6** [comment]
@theus - these look like real errors (I think they were also there in a previous pull request) - maybe @dmitrijus would understand the error?

**Comment by dmitrijus** [comment]
please test

I don't. Restarted tests, let's see if this happens again.
Alternatively, just run the test manually, in your area:
`runTheMatrix -l 25.0`

**Comment by threus** [comment]
So now tests 1000.0 and 25.0 passed, while new others failed. Strange. Which test should I run to debug the problem, if at all? The PR is really simple: booking and filling a 2D plot with bin labels. Nothing fancy.

**Comment by dmitrijus** [comment]
How big is the histogram you are creating? (in number of bins)
I will ping a few people, maybe they will have an idea...

**Comment by dmitrijus** [comment]
This is the reason for failing: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-14793/13432/runTheMatrix-results/1001.0_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3+ALCAHARVD4+ALCAHARVD5/step2_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3+ALCAHARVD4+ALCAHARVD5.log

**Comment by threus** [comment]
@dmitrijus the histo is 450 x 15 bins, so 6750 bins in total. I tested 1000.0 without setBinLabel and it passed. With setBinLabel failed, so it's this piece. But I don't understand why it's a problem and only in some workflows.

File: DQM/SiStripMonitorHardware/src/FEDHistograms.cc [hunk 1]
```diff
@@ -369,6 +415,28 @@ void FEDHistograms::bookTopLevelHistograms(DQMStore::IBooker & ibooker , std::st
 		  "APV-ID",
 		  "FED-ID");
 
+  book2DHistogram( ibooker , fedErrorsVsId_,"FEDErrorsVsId",
```
Comment (dmitrijus) [inline DQM/SiStripMonitorHardware/src/FEDHistograms.cc]: Please check that fedErrorsVsId_ is initialized. book2DHistogram has a nasty if statement...
I suspect in some cases it does not properly book the element and causes the segfault you are receiving.

**Comment by dmitrijus** [comment]
I've made a comment - check the log I linked, it contains the crash.

**Comment by threus** [comment]
@dmitrijus thanks for the hint! Indeed it was not initialised! Apologies. Updated.

**Comment by boudoul** [comment]
Hi @dmitrijus , can you retrigger the tests ? Thanks

**Comment by dmitrijus** [comment]
please test

**Comment by boudoul** [comment]
Hi @threus , looks like there is still something to be fixed...

**Comment by threus** [comment]
I cannot reproduce neither of the failed tests locally with 8_1_0_pre6+this branch. All of them finish successfully with runTheMatrix. I even checked that the plot is in the corresponding DQM file...

**Comment by threus** [comment]
All the above tests are successful also when I run them in the latest available IB, CMSSW_8_1_X_2016-06-27-1100...
Is there something special in the settings of the automated tests that I might miss or need to set up manually?

**Comment by dmitrijus** [comment]
please test

**Comment by threus** [comment]
Apologies again, forgot to commit one line. That's why my tests were all successful and here not. Fixed.

**Comment by boudoul** [comment]
ok so @dmitrijus , could you please restart the test which hopefully should be fine now .
@threus - I assume the corresponding  PR #14794 for 80X is up to date , right?

**Comment by threus** [comment]
@boudoul yep, it is.

**Comment by boudoul** [comment]
Hi @dmitrijus , could you please retrigger the tests (also for 80X ..)

**Comment by boudoul** [comment]
@dmitrijus please....

**Comment by davidlange6** [comment]
please test

________________________________

**Comment by threus** [comment]
thanks @davidlange6 !
@dmitrijus, now that there is only a DAS error, should you retrigger the test or the crash can be safely ignored?

**Comment by boudoul** [comment]
@dmitrijus PING

**Comment by davidlange6** [comment]
please test

**Comment by dmitrijus** [comment]
+1

**Comment by threus** [comment]
thanks all!

**Comment by davidlange6** [comment]
+1

**Pull request merged**